Fix zero return imports, and support for result types

mangas opened the pull request:
- Bumped tool versions of erlang and elixir, hope that's ok
- Added package.json and company to gitignore 
- Functions with no return values were not working correctly, added the native code changes to support this use case. 
- Result types were not being mapped when running running an export. Wire this and added some test coverage. 

The imports test already had most of what I needed so seemed like a good idea to add more cases to it.

mangas commented:
Hey @tessi, just in case you missed the PR, can I get a review on this when you get the chance?

superchris commented:
Hey, I'm sorry I didn't notice this PR before I added result type support in a separate PR. Mine adds support for both inbound and outbound result types, so I'm inclined to keep it, but would happy to have your thoughts on it: #730 . Everything else in this PR looks good tho

tessi commented on .tool-versions:
```diff
@@ -1,3 +1,3 @@
-erlang 27.1.2
-elixir 1.17.3-otp-27
+erlang 27.2.4
+elixir 1.18.2-otp-27
```
💜

tessi commented:
hey @mangas - sorry for letting you wait 💚 and thanks a lot for you PR!

mangas commented:
> Hey, I'm sorry I didn't notice this PR before I added result type support in a separate PR along with the other remaining types other than resources. Everything else looks good, sorry for causing conflicts for you :(

All good, I've fixed the conflicts now

mangas commented:
> hey @mangas - sorry for letting you wait 💚 and thanks a lot for you PR!

No worries at all, I've rebased the PR, let me know if there's any changes needed

tessi commented on native/wasmex/src/component_instance.rs:
```diff
@@ -341,6 +343,10 @@ fn populate_return_values(
 
     let wit_type = match results {
         wit_parser::Results::Anon(wit_type) => wit_type,
+        wit_parser::Results::Named(vec) if vec.is_empty() => {
+            *return_values = Some((true, vals));
+            return Ok(());
+        }
```
❤️

tessi commented:
Hey @mangas - thanks again for your contribution! the code looks great and I love the tests you added. 💚
